dataclients/kubernetes/clusterclient.go

```diff
@@ -283,16 +291,35 @@ func (c *clusterClient) loadIngresses() ([]*ingressItem, error) {
 	return fItems, nil
 }
 
+// skipRouteGroup decides whether a found RouteGroup should be skipped or not.
+func (c *clusterClient) skipRouteGroup(i *routeGroupItem) bool {
```

Comment:
it would be more expressive if the validation and the filtering would stay separate. The name skipRouteGroup doesn't nicely express what it does. I think the two statements just could go inline.